### PR TITLE
Adding Android Whale

### PR DESCRIPTION
**Pull request by laziel**
References https://github.com/browscap/browscap/issues/1917

This adds Android patterns for the Whale browser with tests.

### CODE REVIEW

File: resources/user-agents/browsers/whale/whale.json [hunk 1]
```diff
@@ -79,6 +79,33 @@
...
+        "children": [
+          {
+            "match": "Mozilla/5.0 (#PLATFORM#) AppleWebKit/* (*khtml*like*gecko*) Chrome/* Whale/#MAJORVER#.#MINORVER#* Mobile Safari/*",
+            "device": "general Mobile Phone (Touch)",
+            "platforms": [
+              "Android"
+            ]
+          },
+          {
+            "match": "Mozilla/5.0 (#PLATFORM##DEVICE#) AppleWebKit/* (*khtml*like*gecko*) Chrome/* Whale/#MAJORVER#.#MINORVER#* Mobile Safari/*",
```
Comment (mimmi20) [inline resources/user-agents/browsers/whale/whale.json]: Please remove this rule.

The `#DEVICE#` token without defined devices leads to errors. Without this token its the same as the rule above.

File: resources/user-agents/browsers/whale/whale.json [hunk 2]
```diff
@@ -79,6 +79,33 @@
...
+          "Parent": "DefaultProperties",
+          "Comment": "Whale Browser #MAJORVER#.#MINORVER#",
+          "Version": "#MAJORVER#.#MINORVER#"
+        },
+        "children": [
+          {
+            "match": "Mozilla/5.0 (#PLATFORM#) AppleWebKit/* (*khtml*like*gecko*) Chrome/* Whale/#MAJORVER#.#MINORVER#* Mobile Safari/*",
+            "device": "general Mobile Phone (Touch)",
+            "platforms": [
+              "Android"
```
Comment (mimmi20) [inline resources/user-agents/browsers/whale/whale.json]: Please add `Android_6_0` here too.

File: resources/user-agents/browsers/whale/whale.json [hunk 3]
```diff
@@ -79,6 +79,33 @@
...
+        "engine": "Blink",
+        "device": "general Mobile Device (Touch)",
+        "properties": {
+          "Parent": "DefaultProperties",
+          "Comment": "Whale Browser #MAJORVER#.#MINORVER#",
+          "Version": "#MAJORVER#.#MINORVER#"
+        },
+        "children": [
+          {
+            "match": "Mozilla/5.0 (#PLATFORM#) AppleWebKit/* (*khtml*like*gecko*) Chrome/* Whale/#MAJORVER#.#MINORVER#* Mobile Safari/*",
```
Comment (mimmi20) [inline resources/user-agents/browsers/whale/whale.json]: Please remove the version tokens (`#MAJORVER#` and `#MINORVER#`) here.

File: resources/user-agents/browsers/whale/whale.json [hunk 4]
```diff
@@ -79,6 +79,33 @@
...
+    },
+    {
+        "userAgent": "Whale Browser for Android",
+        "browser": "whale browser",
+        "platform": "Android",
+        "engine": "Blink",
+        "device": "general Mobile Device (Touch)",
+        "properties": {
+          "Parent": "DefaultProperties",
+          "Comment": "Whale Browser #MAJORVER#.#MINORVER#",
```
Comment (mimmi20) [inline resources/user-agents/browsers/whale/whale.json]: Please remove the version tokens (`#MAJORVER#` and `#MINORVER#`) here.

File: resources/user-agents/browsers/whale/whale.json [hunk 5]
```diff
@@ -79,6 +79,33 @@
...
+          "Version": "#MAJORVER#.#MINORVER#"
```
Comment (mimmi20) [inline resources/user-agents/browsers/whale/whale.json]: Please remove this Property.

File: tests/issues/issue-1917.php [hunk 6]
```diff
@@ -130,4 +130,47 @@
         'standard' => true,
         'full' => true,
     ],
+    'issue-1917-D' => [
```
Comment (mimmi20) [inline tests/issues/issue-1917.php]: Please move this test into in its own file (`issue-1921.php`).

File: resources/user-agents/browsers/whale/whale.json [hunk 7]
```diff
@@ -79,6 +79,33 @@
...
         }
       ]
-    }
+    },
+    {
+        "userAgent": "Whale Browser for Android",
+        "browser": "whale browser",
+        "platform": "Android",
+        "engine": "Blink",
+        "device": "general Mobile Device (Touch)",
```
Comment (mimmi20) [inline resources/user-agents/browsers/whale/whale.json]: Please use `general Mobile Phone (Touch)` as default and remove this from the child.

Comment (laziel) [reply resources/user-agents/browsers/whale/whale.json]: Thank you for your review. I have corrected that you pointed.

File: resources/user-agents/browsers/whale/whale.json (hunk 5 above)
Comment (laziel) [reply resources/user-agents/browsers/whale/whale.json]: Thank you for your review. I have corrected that you pointed.

File: resources/user-agents/browsers/whale/whale.json (hunk 4 above)
Comment (laziel) [reply resources/user-agents/browsers/whale/whale.json]: Thank you for your review. I have corrected that you pointed.

File: resources/user-agents/browsers/whale/whale.json (hunk 3 above)
Comment (laziel) [reply resources/user-agents/browsers/whale/whale.json]: Thank you for your review. I have corrected that you pointed.

File: resources/user-agents/browsers/whale/whale.json (hunk 2 above)
Comment (laziel) [reply resources/user-agents/browsers/whale/whale.json]: Thank you for your review. I have corrected that you pointed.

File: resources/user-agents/browsers/whale/whale.json (hunk 1 above)
Comment (laziel) [reply resources/user-agents/browsers/whale/whale.json]: Thank you for your review. I have corrected that you pointed.

File: tests/issues/issue-1917.php [hunk 8]
```diff
@@ -130,4 +130,47 @@
...
+        'ua' => 'Mozilla/5.0 (Linux; Android 6.0.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.113 Whale/0.9.5.0 Mobile Safari/537.36',
+        'properties' => [
+            'Comment' => 'Whale Browser',
+            'Browser' => 'Whale Browser',
+            'Browser_Type' => 'Browser',
+            'Browser_Bits' => '32',
+            'Browser_Maker' => 'NHN Corporation',
+            'Browser_Modus' => 'unknown',
+            'Version' => '0.9',
```
Comment (mimmi20) [inline tests/issues/issue-1917.php]: Please change the version to `0.0`. This let the test fail.

File: resources/user-agents/browsers/whale/whale.json [hunk 9]
```diff
@@ -79,6 +79,26 @@
...
+        "device": "general Mobile Phone (Touch)",
+        "properties": {
+          "Parent": "DefaultProperties",
+          "Comment": "Whale Browser"
+        },
+        "children": [
+          {
+            "match": "Mozilla/5.0 (#PLATFORM#) AppleWebKit/* (*khtml*like*gecko*) Chrome/* Whale/* Mobile Safari/*",
+            "platforms": [
+              "Android",
```
Comment (mimmi20) [inline resources/user-agents/browsers/whale/whale.json]: Please switch the two versions (`Android` at last)

File: tests/issues/issue-1917.php (hunk 8 above)
Comment (laziel) [reply tests/issues/issue-1917.php]: Changed here to `0.0`. Thank you for your generosity.

File: resources/user-agents/browsers/whale/whale.json (hunk 9 above)
Comment (laziel) [reply resources/user-agents/browsers/whale/whale.json]: Switched the two versions. Thank you for your generosity 👍

**Pull request merged**